bedrock/newsletter/views.py

```diff
@@ -99,9 +98,20 @@ def confirm(request, token):
             qparams.append(qs)
         return HttpResponseRedirect("{}?{}".format(reverse("newsletter.existing.token", kwargs={"token": token}), "&".join(qparams)))
     else:
-        return l10n_utils.render(
-            request, "newsletter/confirm.html", {"success": success, "generic_error": generic_error, "token_error": token_error}, ftl_files=FTL_FILES
-        )
+        qparams = ["error=1"] if generic_error else ["error=2"] if token_error else ["success=1"]
+
+        qs = request.META.get("QUERY_STRING", "")
+
+        if qs:
+            qparams.append(qs)
```

Comment:
Since Python 3.9+ this pattern can be written using the walrus operator, if you prefer it:

```
if qs := request.META.get("QUERY_STRING", ""):
    qparams.append(qs)
```